mava/wrappers/jaxmarl.py

```diff
@@ -407,3 +408,37 @@ def get_global_state(self, wrapped_env_state: BraxState, obs: Dict[str, Array])
         """Get global state from observation and copy it for each agent."""
         # Use the global state of brax.
         return jnp.tile(wrapped_env_state.obs, (self.num_agents, 1))
+
+
+class MPEWrapper(JaxMarlWrapper):
+    """Wrapper for the MPE environment."""
+
+    def __init__(
+        self,
+        env: SimpleSpreadMPE,
+        has_global_state: bool = False,
+    ):
+        super().__init__(env, has_global_state, env.max_steps)
+        self._env: SimpleSpreadMPE
+
+    @cached_property
+    def action_dim(self) -> chex.Array:
+        "Get the actions dim for each agent."
+        # Adjusted automatically based on the action_type specified in the kwargs.
+        if _is_discrete(self._env.action_space(self.agents[0])):
+            return self._env.action_space(self.agents[0]).n
+        return self._env.action_space(self.agents[0]).shape[0]
+
+    @cached_property
+    def state_size(self) -> chex.Array:
+        "Get the state size of the global observation"
+        return self._env.observation_space(self.agents[0]).shape[0] * self.num_agents
+
```

Comment:
```suggestion

    def observation_spec(self) -> specs.Spec:
        """Returns the observation spec."""
        # JaxMARL `env.observation_spaces` don't give the correct shape for MPE with more than 3 agent.
        # We need to get the correct shape by creating a mock state and getting the shape from it.
        mock_state=self._env.reset(jax.random.PRNGKey(0))[0]
        observation_shape = mock_state["agent_0"].shape
        observation_spaces = self._env.observation_spaces
        for i in range(self.num_agents):
            observation_spaces[f"agent_{i}"].shape = observation_shape
        agents_view = jaxmarl_space_to_jumanji_spec(merge_space(observation_spaces))

        action_mask = specs.BoundedArray(
            (self.num_agents, self.action_dim), bool, False, True, "action_mask"
        )
        step_count = specs.BoundedArray(
            (self.num_agents,), jnp.int32, 0, self.time_limit, "step_count"
        )

        if self.has_global_state:
            global_state = specs.Array(
                (self.num_agents, observation_spaces["agent_0"].shape[0]*self.num_agents),
                agents_view.dtype,
                "global_state",
            )

            return specs.Spec(
                ObservationGlobalState,
                "ObservationSpec",
                agents_view=agents_view,
                action_mask=action_mask,
                global_state=global_state,
                step_count=step_count,
            )

        return specs.Spec(
            Observation,
            "ObservationSpec",
            agents_view=agents_view,
            action_mask=action_mask,
            step_count=step_count,
        )


```

Reply:
MPE with more than 3 agents is broken because of the original generating of obs space so I suggested this function to fix the issue

Reply:
I removed the "state_size" fn since we don't need it anymore

Reply:
We no longer need to to get a `mock_obs` and `reset` env since this is fixed in the new jaxmarl version that we agreed to be using 😄

Reply:
Agreed we shouldn't need this because we are now using the newest jaxmarl and they should have fixed this

Reply:
Ah valid I tested it now 😅